src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
+    const length = this.length;
+
+    for (let i = 0; i < this.length; i++) {
+      if (typeof this[i] === 'undefined'
+      || this[i] === null) {
+        this[i] = '';
+      }
```

Comment:
Why don't you just add `this[i]` to string if it's not `null` or undefined, you're repeating yourself in next `if` block - try to invert your if statement to do that. Also, you don't need `typeof` here.